File: firebase/functions/src/model/PlayerDuel.ts
import { DocumentReference, Timestamp } from "firebase-admin/firestore";

export type PlayerDuel = {
    firstPlayer: DocumentReference,
    firstPlayerBoard: string,
    secondPlayer: DocumentReference,
    secondPlayerBoard: string,
    given: string,
    startTime: Timestamp,
    sudoku: DocumentReference,
    winner: DocumentReference | null
}

File: app/ios/Sudoku Battles/Data/Model/Firebase/PlayerDuel.swift
import Foundation
import FirebaseFirestore

struct PlayerDuel: Codable {
    let firstPlayer: DocumentReference
    let firstPlayerBoard: String
    let secondPlayer: DocumentReference
    let secondPlayerBoard: String
    let given: String
    let startTime: Timestamp
    let endTime: Timestamp?
    let sudoku: DocumentReference
    let winner: DocumentReference?
    
    var difficulty: Difficulty { Difficulty(rawValue: sudoku.parent.collectionID) ?? .easy }
}
